### PR TITLE
Add ACE editor on submission

### PR DESCRIPTION
**Pull request by alkeshrails**

### CODE REVIEW

File: app/views/submissions/new/_form.html.erb [hunk 1]
```diff
@@ -6,40 +6,58 @@
...
     <% end %>
-  </div>
+
+    <div class='btn-set md-content'>
+      <%= f.submit 'Submit Submission', class: 'btn btn-primary btn-sm' %>
+    </div>
+  <% end %>
+</div>
+
+<script>
```
Comment (piotrekpasciak) [inline app/views/submissions/new/_form.html.erb]: Could you move this script to Stimulus controller?

Example controller:
https://github.com/AIcrowd/AIcrowd/blob/master/app/javascript/stimulus_js/controllers/chart_controller.js
Example usage:
https://github.com/AIcrowd/AIcrowd/blob/master/app/views/insights/_charts/_top_score_vs_time.html.erb#L1-L7

File: db/migrate/20200330140331_add_code_submissions_to_challenge.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddCodeSubmissionsToChallenge < ActiveRecord::Migration[5.2]
+  def change
+    add_column :challenges, :code_submissions, :boolean, default: false
```
Comment (piotrekpasciak) [inline db/migrate/20200330140331_add_code_submissions_to_challenge.rb]: It's good practice to not allow `NULL` value in database when they're not needed. Please add `null: false` constraint.

File: spec/features/challenges/submission_spec.rb [hunk 3]
```diff
@@ -50,11 +50,17 @@
     visit new_challenge_submission_path(running)
     expect(page).to have_text 'Create Submission'
   end
+
+  it do
```
Comment (piotrekpasciak) [inline spec/features/challenges/submission_spec.rb]: Add description what this spec does `it 'does something' do`

File: spec/features/challenges/submission_spec.rb [hunk 4]
```diff
@@ -66,6 +72,12 @@
     expect(page).to have_text 'This challenge is now completed. You may continue to make submissions and your entries will appear on the Ongoing Leaderboard.'
   end
 
+  it do
```
Comment (piotrekpasciak) [inline spec/features/challenges/submission_spec.rb]: Add description here as well.

**Comment by skbly7** [comment]
Merged as part of https://github.com/AIcrowd/AIcrowd/pull/1361

**Pull request closed without merging**